src/Toml/BiMap.hs

```diff
@@ -51,245 +55,270 @@ module Toml.BiMap
 
 -- | 'Bool' bimap for 'AnyValue'. Usually used with 'bool' combinator.
-_Bool :: BiMap Bool AnyValue
+_Bool :: TomlBiMap Bool AnyValue
 _Bool = mkAnyValueBiMap matchBool Bool
 
 -- | 'Integer' bimap for 'AnyValue'. Usually used with 'integer' combinator.
-_Integer :: BiMap Integer AnyValue
+_Integer :: TomlBiMap Integer AnyValue
 _Integer = mkAnyValueBiMap matchInteger Integer
 
 -- | 'Double' bimap for 'AnyValue'. Usually used with 'double' combinator.
-_Double :: BiMap Double AnyValue
+_Double :: TomlBiMap Double AnyValue
 _Double = mkAnyValueBiMap matchDouble Double
 
 -- | 'Text' bimap for 'AnyValue'. Usually used with 'text' combinator.
-_Text :: BiMap Text AnyValue
+_Text :: TomlBiMap Text AnyValue
 _Text = mkAnyValueBiMap matchText Text
 
 -- | Zoned time bimap for 'AnyValue'. Usually used with 'zonedTime' combinator.
-_ZonedTime :: BiMap ZonedTime AnyValue
+_ZonedTime :: TomlBiMap ZonedTime AnyValue
 _ZonedTime = mkAnyValueBiMap (matchDate >=> getTime) (Date . Zoned)
   where
-    getTime (Zoned z) = Just z
-    getTime _         = Nothing
+    getTime (Zoned z) = Right z
+    getTime value     = leftWrongValue $ Date value
 
 -- | Local time bimap for 'AnyValue'. Usually used with 'localTime' combinator.
-_LocalTime :: BiMap LocalTime AnyValue
+_LocalTime :: TomlBiMap LocalTime AnyValue
 _LocalTime = mkAnyValueBiMap (matchDate >=> getTime) (Date . Local)
   where
-    getTime (Local l) = Just l
-    getTime _         = Nothing
+    getTime (Local l) = Right l
+    getTime value     = leftWrongValue $ Date value
 
 -- | Day bimap for 'AnyValue'. Usually used with 'day' combinator.
-_Day :: BiMap Day AnyValue
+_Day :: TomlBiMap Day AnyValue
 _Day = mkAnyValueBiMap (matchDate >=> getTime) (Date . Day)
   where
-    getTime (Day d) = Just d
-    getTime _       = Nothing
+    getTime (Day d) = Right d
+    getTime value   = leftWrongValue $ Date value
 
 -- | Time of day bimap for 'AnyValue'. Usually used with 'timeOfDay' combinator.
-_TimeOfDay :: BiMap TimeOfDay AnyValue
+_TimeOfDay :: TomlBiMap TimeOfDay AnyValue
 _TimeOfDay = mkAnyValueBiMap (matchDate >=> getTime) (Date . Hours)
   where
-    getTime (Hours h) = Just h
-    getTime _         = Nothing
+    getTime (Hours h) = Right h
+    getTime value     = leftWrongValue $ Date value
 
 -- | Helper bimap for 'String' and 'Text'.
-_StringText :: BiMap String Text
+_StringText :: BiMap e String Text
 _StringText = iso T.pack T.unpack
 
 -- | 'String' bimap for 'AnyValue'. Usually used with 'string' combinator.
-_String :: BiMap String AnyValue
+_String :: TomlBiMap String AnyValue
 _String = _StringText >>> _Text
 
 -- | Helper bimap for 'String' and types with 'Read' and 'Show' instances.
-_ReadString :: (Show a, Read a) => BiMap a String
-_ReadString = BiMap (Just . show) readMaybe
+_ReadString :: (Show a, Read a) => TomlBiMap a String
+_ReadString = BiMap (Right . show)
+    (\value -> either (\_ -> leftWrongValue . Text $ toTxt value) Right . readEither $ value)
```

Comment:
It's good idea to keep error returned by `readEither` function and tell this error.